From 35d2a5c9ff6fb8c9cc54fc6c09a0423630b27cbd Mon Sep 17 00:00:00 2001
From: mcop1 <marco.perberschlager@pimcore.com>
Date: Wed, 20 Jul 2022 11:50:00 +0200
Subject: [PATCH 1/2] [Bug]: thumbnail resolutions multiple types list doesn't
 work anymore Check for value kind and use the correct property

---
 src/GraphQL/FieldHelper/AbstractFieldHelper.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/GraphQL/FieldHelper/AbstractFieldHelper.php b/src/GraphQL/FieldHelper/AbstractFieldHelper.php
index 61724672..aa57aee4 100644
--- a/src/GraphQL/FieldHelper/AbstractFieldHelper.php
+++ b/src/GraphQL/FieldHelper/AbstractFieldHelper.php
@@ -100,7 +100,7 @@ public function getArguments(FieldNode $ast)
         for ($i = 0; $i < $count; $i++) {
             /** @var ArgumentNode $argumentNode */
             $argumentNode = $nodeList[$i];
-            $value = $argumentNode->value->value;
+            $value = $argumentNode->value->kind === "ListValue" ? $argumentNode->value->values : $argumentNode->value->value;
             $result[$argumentNode->name->value] = $value;
         }
 

From da89db3b5db5fd8821194a9fe9b21a45ef2f379a Mon Sep 17 00:00:00 2001
From: mcop1 <mcop1@users.noreply.github.com>
Date: Wed, 20 Jul 2022 09:50:25 +0000
Subject: [PATCH 2/2] Apply php-cs-fixer changes

---
 src/GraphQL/FieldHelper/AbstractFieldHelper.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/GraphQL/FieldHelper/AbstractFieldHelper.php b/src/GraphQL/FieldHelper/AbstractFieldHelper.php
index aa57aee4..82adb6cc 100644
--- a/src/GraphQL/FieldHelper/AbstractFieldHelper.php
+++ b/src/GraphQL/FieldHelper/AbstractFieldHelper.php
@@ -100,7 +100,7 @@ public function getArguments(FieldNode $ast)
         for ($i = 0; $i < $count; $i++) {
             /** @var ArgumentNode $argumentNode */
             $argumentNode = $nodeList[$i];
-            $value = $argumentNode->value->kind === "ListValue" ? $argumentNode->value->values : $argumentNode->value->value;
+            $value = $argumentNode->value->kind === 'ListValue' ? $argumentNode->value->values : $argumentNode->value->value;
             $result[$argumentNode->name->value] = $value;
         }
 
